src/Exception/GoogleException.php

```diff
@@ -0,0 +1,23 @@
+<?php
+/**
+ * Copyright 2016 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+namespace Google\Cloud\Exception;
+
+class GoogleException extends \Exception
```

Comment:
For cases that don't fall under normal [SPL exception](http://php.net/manual/en/spl.exceptions.php) use cases I thought it could be useful to bubble up library specific exceptions so users could get a better indication of how to handle errors. It definitely isn't required, but could provide helpful.